src/Cli/dotnet/commands/dotnet-workload/install/FileBasedInstaller.cs

```diff
@@ -452,21 +452,34 @@ public void GarbageCollect(Func<string, IWorkloadResolver> getResolverForWorkloa
 
         }
 
-        public void DeleteInstallState(SdkFeatureBand sdkFeatureBand)
+        public void RemoveManifestsFromInstallState(SdkFeatureBand sdkFeatureBand)
         {
             string path = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetDir), "default.json");
             
             if (File.Exists(path))
             {
-                File.Delete(path);
+                var installStateContents = InstallStateContents.FromString(File.Exists(path) ? File.ReadAllText(path) : "{}");
```

Comment:
If the file doesn't exist, I think it's better to just create a new `InstallStateContents` object directly rather than parsing an empty JSON string.  See also other cases of this later in the file.

```suggestion
                var installStateContents = File.Exists(path) ? InstallStateContents.FromString(File.ReadAllText(path)) : new InstallStateContents();
```